airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -258,15 +262,32 @@ def error_message(self, response: requests.Response) -> str:
         """
         return ""
 
+    def must_deduplicate_query_params(self) -> bool:
+        return False
+
+    def deduplicate_query_params(self, url: str, params: Optional[Mapping[str, Any]]) -> Mapping[str, Any]:
+        if params is None:
+            params = {}
+        query_string = urllib.parse.urlparse(url).query
+        query_dict = {k: v[0] for k, v in urllib.parse.parse_qs(query_string).items()}
+
+        duplicate_keys_with_same_value = {k for k in query_dict.keys() if params.get(k) == query_dict[k]}
```

Comment:
As `urllib.parse.parse_qs` doesn't differentiate between strings and numbers (because that's not a concept for query params), this check fails if the query param in question is a number (or anything besides a string)
<img width="1268" alt="Screenshot 2023-07-24 at 10 35 08" src="https://github.com/airbytehq/airbyte/assets/1508364/12e05fb8-b605-4406-a218-a2019c862d4a">

In this example, q is not de-duplicated

Reply:
good catch! Fixed by casting the values to strings